src/components/organisms/edit-denominations-modal/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react"
+import { v4 as uuidv4 } from "uuid"
+import Button from "../../fundamentals/button"
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import TrashIcon from "../../fundamentals/icons/trash-icon"
+import InfoTooltip from "../../molecules/info-tooltip"
+import Modal from "../../molecules/modal"
+import CurrencyInput from "../../organisms/currency-input"
+
+export type PriceType = {
+  currency_code: string
+  amount: number
+  id?: string
+}
+
+type EditDenominationsModalProps = {
+  defaultDenominations: PriceType[]
+  handleClose: () => void
+  onSubmit: (denominations: PriceType[]) => void
+  defaultNewAmount?: number
+  defaultNewCurrencyCode?: string
+  currencyCodes?: string[]
+}
+
+const augmentWithId = (obj) => ({ ...obj, id: obj.id ? obj.id : uuidv4() })
```

Comment:
This tampers a bit with the data when we send the denominations to the backend because we are providing the id ourselves instead of letting Postgres generate it.

We could say, that the parent component is required to normalize the data from the denominations modal to fit the backend schema, but that would not be very obvious to developers coming from the outside using the component. Maybe we could add a different prop `indexId` that is dedicated to solving the rendering issue and then remove that prop on all denominations when submitting the data? 

Let me know what you think 👍

Reply:
great idea oli :+1: !